app/routes/articles.js

```diff
@@ -95,13 +99,13 @@ router.get('/approve/:id', isLoggedIn, async (req, res) => {
 })
 
 //APPROVE Approve Article Route
-router.post('/approve/:id', isLoggedIn, (req, res) => {
-    if(!req.user.isAdmin || !req.params.id) {
+router.post('/approve/:link', isLoggedIn, (req, res) => {
+    if(!req.user.isAdmin || !req.params.link) {
         req.flash('error', 'Oops! Something went wrong!')
         return res.redirect('/')
     }
 
-    Article.findOne({_id: req.params.id}, (err, article) => {
+    Article.findOne({link: encodeURIComponent(req.params.link)}, (err, article) => {
```

Comment:
It's not fully clear to me why, sometimes you both replace spaces with dashes and encode URL, and sometimes you just encode URL without spaces replacement.

Reply:
After link creation it already has the dashes, therefore there's no reason to replace spaces with dashes.

Reply:
Because in this instance, there should not be spaces, if there is, it's an incorrect url

Reply:
Why do we encode req.params.link which is already a working link